generated-source-files.mk

```diff
@@ -0,0 +1,139 @@
+gorename_package         := github.com/prateek/gorename
+gorename_package_version := 52c7307cddd221bb98f0a3215216789f3c821b10
+
+# Tests that all currently generated types match their contents if they were regenerated
+.PHONY: test-genny-all
+test-genny-all: genny-all
+	@test "$(shell git diff --shortstat 2>/dev/null)" = "" || (git diff --no-color && echo "Check git status, there are dirty files" && exit 1)
+	@test "$(shell git status --porcelain 2>/dev/null | grep "^??")" = "" || (git status --porcelain && echo "Check git status, there are untracked files" && exit 1)
+
+.PHONY: genny-all
+genny-all: genny-map-all genny-arraypool-all
+
+.PHONY: genny-map-all
+genny-map-all: idhashmap-update byteshashmap-update
+
+.PHONY: idhashmap-update
+idhashmap-update: install-generics-bin
+	cd generics/hashmap && cat ./map.go | grep -v nolint | genny -pkg idkey gen "KeyType=ident.ID ValueType=MapValue" > ./idkey/map_gen.go
+
+.PHONY: byteshashmap-update
+byteshashmap-update: install-generics-bin
+	cd generics/hashmap && cat ./map.go | grep -v nolint | genny -pkg byteskey gen "KeyType=[]byte ValueType=MapValue" > ./byteskey/map_gen.go
+
+# generic targets meant to be re-used by other users
+.PHONY: hashmap-gen
+hashmap-gen: install-generics-bin
+	$(eval out_dir=$(gopath_prefix)/$(target_package))
+	cd generics/hashmap && cat ./map.go | grep -v nolint | genny -pkg $(pkg) gen "KeyType=$(key_type) ValueType=$(value_type)" > "$(out_dir)/map_gen.go"
+ifneq ($(rename_type_prefix),)
+	make hashmap-gen-rename
+endif
+
+.PHONY: idhashmap-gen
+idhashmap-gen: install-generics-bin
+	$(eval out_dir=$(gopath_prefix)/$(target_package))
+	cd generics/hashmap/idkey && cat ./map_gen.go | grep -v nolint | genny -pkg $(pkg) gen "MapValue=$(value_type)" > "$(out_dir:\=)/map_gen.go"
+	cd generics/hashmap/idkey && cat ./new_map.go | grep -v nolint | genny -pkg $(pkg) gen "MapValue=$(value_type)" > "$(out_dir:\=)/new_map_gen.go"
+ifneq ($(rename_type_prefix),)
+	make hashmap-gen-rename
+endif
+
+.PHONY: byteshashmap-gen
+byteshashmap-gen: install-generics-bin
+	$(eval out_dir=$(gopath_prefix)/$(target_package))
+	cd generics/hashmap/byteskey && cat ./map_gen.go | grep -v nolint | genny -pkg $(pkg) gen "MapValue=$(value_type)" > "$(out_dir:\=)/map_gen.go"
+	cd generics/hashmap/byteskey && cat ./new_map.go | grep -v nolint | genny -pkg $(pkg) gen "MapValue=$(value_type)" > "$(out_dir:\=)/new_map_gen.go"
+ifneq ($(rename_type_prefix),)
+	make hashmap-gen-rename
+endif
+
+.PHONY: hashmap-gen-rename-helper
+hashmap-gen-rename-helper:
+	gorename -from '"$(target_package)$(temp_suffix)".Map' -to $(rename_type_prefix)Map
```

Comment:
Can we declare a fixed `temp_suffix` so callers don't need to set it?  Also assume probably want to use `_tmp` instead of just `tmp` otherwise it'll be some strange target package like `../storage/repairtemp/...` rather than `../storage/repair_tmp/...`.

Can you also add a comment somewhere that `target_package` needs to avoid having a trailing slash?

Reply:
sure thing

Reply:
done